flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/ProjectionTranslator.scala

```diff
@@ -226,30 +226,69 @@ object ProjectionTranslator {
       overWindows: Array[OverWindow],
       tEnv: TableEnvironment): Seq[Expression] = {
 
-    def resolveOverWindow(unresolvedCall: UnresolvedOverCall): Expression = {
-
-      val overWindow = overWindows.find(_.alias.equals(unresolvedCall.alias))
-      if (overWindow.isDefined) {
-        OverCall(
-          unresolvedCall.agg,
-          overWindow.get.partitionBy,
-          overWindow.get.orderBy,
-          overWindow.get.preceding,
-          overWindow.get.following)
-      } else {
-        unresolvedCall
-      }
-    }
+    exprs.map(e => replaceOverCall(e, overWindows, tEnv))
+  }
 
-    val projectList = new ListBuffer[Expression]
-    exprs.foreach {
-      case Alias(u: UnresolvedOverCall, name, _) =>
-        projectList += Alias(resolveOverWindow(u), name)
+  /**
+    * Find and replace UnresolvedOverCall with OverCall
+    *
+    * @param expr    the expression to check
+    * @return an expression with correct resolved OverCall
+    */
+  private def replaceOverCall(
+    expr: Expression,
+    overWindows: Array[OverWindow],
+    tableEnv: TableEnvironment): Expression = {
+
+    expr match {
       case u: UnresolvedOverCall =>
-        projectList += resolveOverWindow(u)
-      case e: Expression => projectList += e
+        val overWindow = overWindows.find(_.alias.equals(u.alias))
+        if (overWindow.isDefined) {
+          OverCall(
+            u.agg,
+            overWindow.get.partitionBy,
+            overWindow.get.orderBy,
+            overWindow.get.preceding,
+            overWindow.get.following)
+        } else {
+          u
+        }
+
+      case u: UnaryExpression =>
+        val c = replaceOverCall(u.child, overWindows, tableEnv)
```

Comment:
Ah, I meant the current test case can not execute to this branch， I suggest add `Abs((countFun('b) over 'w))` into test case SELECT clause. i.e:
 .select(
        'a, 'b, 'c,
        'b.sum over 'w,
        "SUM:".toExpr + ('b.sum over 'w),
         countFun('b) over 'w,
         (countFun('b) over 'w) + 1, 
         **Abs((countFun('b) over 'w)), -> // cover UnaryExpression case**
         plusOne(countFun('b) over 'w),
         **array(countFun('c) over 'w, countFun('b) over 'w), -> // cover ArrayConstructor case**
        'b.avg over 'w,
        'b.max over 'w,
        'b.min over 'w,
        weightAvgFun('b, 'a) over 'w)